src/sentry/sentry_metrics/querying/metadata/metrics_code_locations.py

```diff
@@ -99,26 +99,31 @@ def __init__(
 
         self._redis_client = get_redis_client_for_metrics_meta()
 
-        self._validate()
-
-    def _validate(self):
-        total_combinations = len(self._projects) * len(self._metric_mris) * len(self._timestamps)
-        if total_combinations > self.MAXIMUM_KEYS:
-            raise TooManyCodeLocationsRequestedError(
-                "The request results in too many code locations to be fetched, try to reduce the number of "
-                "metrics, projects or the time interval"
-            )
-
     def _code_location_queries(self) -> Generator[CodeLocationQuery, None, None]:
+        total_count = len(self._projects) * len(self._metric_mris) * len(self._timestamps)
+        step_size = (
+            1 if total_count <= self.MAXIMUM_KEYS else math.ceil(total_count / self.MAXIMUM_KEYS)
+        )
+
+        # We want to distribute evenly and deterministically the elements in the set of combinations. For example, if
+        # the total count of code locations queries you made is 100 and our maximum is 50, then we will sample 1 out of
+        # 2 elements out of the 100 queries, to be within the 50.
+        current_step = 0
         for project in self._projects:
             for metric_mri in self._metric_mris:
                 for timestamp in self._timestamps:
-                    yield CodeLocationQuery(
-                        organization_id=self._organization.id,
-                        project_id=project.id,
-                        metric_mri=metric_mri,
-                        timestamp=timestamp,
-                    )
+                    if current_step == step_size:
+                        current_step = 0
+
+                    if current_step == 0:
```

Comment:
super nit: I would have used the modulo operator here to have less logical branches.
```suggestion
                    if current_step % step_size == 0:
```

Reply:
I was also considering it, just didn't want to divide since with a branch it was avoided but let's opt for elegance.